### PR TITLE
Rename BeginScope to BeginScopeImpl in favor of using the BeginScope log...

### PR DESCRIPTION
**Pull request by kichalla**
...ger extension method.

This change will probably effect other repos too, so will make changes there too.

@rynowak @suhasj @loudej

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@kichalla__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Kiran Challa). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

File: samples/SampleApp/Program.cs [hunk 1]
```diff
@@ -63,7 +63,7 @@ public void Main(string[] args)
                 _logger.LogWarning("Unexpected warning", ex);
             }
 
-            using (_logger.BeginScope("Main"))
+            using (_logger.BeginScopeImpl("Main"))
```
Comment (kichalla) [inline samples/SampleApp/Program.cs]: Will change this sample (and others...like in Entropy) to use `BeginScope` logger extension once PR https://github.com/aspnet/Logging/pull/159 is checked in.

**Comment by Eilon** [comment]
I assume this was something that was discussed with @lodejard ? The code looks :shipit: to me.

**Comment by rynowak** [comment]
This rename was the best option we could come up with in the logging hackathon the other day. :laughing: 

The intent is to still define the interface in a simple way, but make calling this method directly unattractive.

**Comment by Eilon** [comment]
Yeah you definitely achieved those goals... very unattractive :smile:

**Comment by kichalla** [comment]
commit d072cde33ec988e879504232bd9c43bd75a9a9be

**Comment by avanderhoorn** [comment]
This might just be me but `BeginScopeImpl` is really ugly. For instance, at first glance I'm not even sure what "Impl" means... If its something you don't want people to really use, surely there is a better way of doing this.

**Comment by tugberkugurlu** [comment]
I 100% agree with @avanderhoorn here. Why is this prefixed with `Impl`? Looking at the PR and changes but I cannot understand why and it looks like this is a member which should not be public but it's even worse: it's on the interface :smile:

**Comment by Eilon** [comment]
We're certainly open to suggestions on better ways of doing this... part of the idea is that most consumers of this interface will see the extension methods alongside the interface method and will call the "prettiest" and "most obviously correct" methods, and not some ugly "XyzImpl" method.

**Pull request closed without merging**